faunadb-java-dsl/src/main/java/com/faunadb/client/types/Value.java

```diff
@@ -82,7 +84,55 @@ private Value() {
    * @see Codec
    */
   public final <T> Result<T> to(Codec<T> codec) {
-    return codec.apply(this);
+    return codec.decode(this);
+  }
+
+  /**
+   * Attempts to coerce this value using the {@link Decoder} class into the specified type
+   *
+   * @param clazz a class type to convert
+   * @return the {@link Result} of the coercion
+   * @see Decoder
+   */
+  public final <T> Result<T> to(Class<T> clazz) {
+    return Decoder.decode(this, Types.of(clazz));
+  }
+
+  /**
+   * Attempts to convert the specified object into the corresponding {@link Value} type
+   *
+   * @param obj the object instance to convert
+   * @return the {@link Result} of the conversion
+   */
+  public static <T> Result<Value> from(T obj) {
+    return Encoder.encode(obj);
+  }
+
+  /**
+   * Attempts to coerce this value to a {@link Map}.
+   * This method is only util if this object is an instance of {@link ObjectV}.
+   *
+   * @param keyType the type of the keys. Currently only {@link String} types are supported.
+   * @param valueType the type of the values.
+   * @return a {@link Result} with the resulting map containing the keys/values.
+   * @see Decoder
+   * @see Types
+   */
+  public final <K, V> Result<Map<K, V>> asMapOf(Class<K> keyType, Class<V> valueType) {
```

Comment:
I think that's confusing. People will try to create maps with different key types and will fail at runtime. I'd rather enforce string as keys to keep the contract consistent.